source/core/slang-char-util.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SLANG_CORE_CHAR_UTIL_H
+#define SLANG_CORE_CHAR_UTIL_H
+
+#include "slang-string.h"
+
+namespace Slang {
+
+struct CharUtil
+{
+    typedef uint8_t Flags;
+    struct Flag
+    {
+        enum Enum : Flags
+        {
+            Upper           = 0x01,
+            Lower           = 0x02,
+            Digit           = 0x04,         
+            LineWhitespace  = 0x08,         ///< Whitespace that can appear in a line (ie excluding CR/LF)
+            HexDigit        = 0x10,         ///< 0-9, a-f, A-F
+        };
+    };
+
+    SLANG_FORCE_INLINE static bool isDigit(char c) { return c >= '0' && c <= '9'; }
+    SLANG_FORCE_INLINE static bool isLower(char c) { return c >= 'a' && c <= 'z'; }
+    SLANG_FORCE_INLINE static bool isUpper(char c) { return c >= 'A' && c <= 'Z'; }
+    SLANG_FORCE_INLINE static bool isLineWhitespace(char c) { return c == ' ' || c == '\t'; }
+
+        /// True if it's alpha
+    SLANG_FORCE_INLINE static bool isAlpha(char c) { return (getFlags(c) & (Flag::Upper | Flag::Lower)) != 0; }
+
+    SLANG_FORCE_INLINE static bool isHexDigit(char c) { return (getFlags(c) & Flag::HexDigit) != 0; }
+
+        /// For a given character get the associated flags
+    SLANG_FORCE_INLINE static Flags getFlags(char c) { return g_charFlagMap.flags[UInt(c)]; }
+
+        /// Given a character return the lower case equivalent 
+    SLANG_FORCE_INLINE static char toLower(char c) { return (c >= 'A' && c <= 'Z') ? (c -'A' + 'a') : c; }
+        /// Given a character return the upper case equivalent
+    SLANG_FORCE_INLINE static char toUpper(char c) { return (c >= 'a' && c <= 'z') ? (c -'a' + 'A') : c; }
+
+    struct CharFlagMap
+    {
+        Flags flags[0x100];
+    };
+
+    static const CharFlagMap g_charFlagMap;
```

Comment:
Given that all of this stuff is very ASCII-specific, should we make this whole thing be `ASCIIUtil` instead of `CharUtil`? At some point I'd like us to have at least basic support for Unicode text in parts of Slang, but at the same time I expect that ASCII-specific utilities like this will be worth keeping around even in that hypothetical future (because things like the C++ extractor can and should always focus on ASCII only).